File: georocket-server-api/src/main/java/io/georocket/storage/StoreCursor.java
package io.georocket.storage;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

/**
 * A cursor that can be used to iterate over chunks in a {@link Store}
 * @since 1.0.0
 * @author Michel Kraemer
 */
public interface StoreCursor {
  /**
   * @return true if there are more items to iterate over
   */
  boolean hasNext();
  
  /**
   * Get the next item from the store
   * @param handler will be called when the item has been retrieved
   */
  void next(Handler<AsyncResult<ChunkMeta>> handler);
  
  /**
   * @return the absolute path to the chunk that has been produced by the last
   * call to {@link #next(Handler)}
   */
  String getChunkPath();
}

File: georocket-server/src/main/kotlin/io/georocket/storage/StoreCursor.kt
package io.georocket.storage

/**
 * A cursor that can be used to iterate over chunks in a [Store]
 * @author Michel Kraemer
 */
interface StoreCursor : Cursor<ChunkMeta> {
  /**
   * Return the absolute path to the chunk that has been produced by the last
   * call to [next]
   */
  val chunkPath: String

  /**
   * Return info about the set of items
   */
  val info: CursorInfo
}
